opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -2083,35 +2084,116 @@ export const elDeleteElements = async (context, user, elements, stixLoadById) =>
+    return value;
+  }
+  const key = attribute.name;
+  // Only native type
+  if (attribute.type === 'string') { // For string, trim by default
+    if (!R.is(String, value)) {
+      throw UnsupportedError('Invalid data type for string preparation', { type: entityType, attribute: key, value });
+    }
+    return value.trim();
+  }
+  // Come from string or native type
+  if (attribute.type === 'date') { // No need to transform, natively supported by elastic
+    if (!R.is(String, value) && !(value instanceof Date) && !moment.isMoment(value)) {
+      throw UnsupportedError('Invalid data type for data preparation', { type: entityType, attribute, value });
+    }
+    return value; // utcDate(value).toDate();
+  }
+  // Come from string or native type
+  if (attribute.type === 'numeric') { // Patch field is string generic so need to be cast to int
+    if (!R.is(String, value) && !R.is(Number, value)) {
+      throw UnsupportedError('Invalid data type for numeric preparation', { type: entityType, attribute, value });
+    }
+    return R.is(Number, value) ? value : Number(value);
+  }
+  // Come from string or native type
+  if (attribute.type === 'boolean') { // Patch field is string generic so need to be cast to boolean
+    if (!R.is(String, value) && (typeof value !== 'boolean')) {
+      throw UnsupportedError('Invalid data type for boolean preparation', { type: entityType, attribute, value });
+    }
+    return typeof value === 'boolean' ? value : value?.toLowerCase() === 'true';
+  }
+  // Come from only in native type, should be prepared earlier
+  if (attribute.type === 'dictionary') { // For dictionary object, prepare inner elements
+    if (!isObject(value)) {
+      throw UnsupportedError('Invalid data type for dictionary preparation', { type: entityType, attribute, value });
+    }
+    const dictionary = {};
+    Object.keys(value).forEach((k) => {
+      const dictValue = value[k];
+      dictionary[k] = R.is(String, dictValue) ? dictValue.trim() : dictValue;
+    });
+    return dictionary;
+  }
+  // Come from string or native type
+  if (attribute.type === 'json') {
+    if (!R.is(String, value) && !isObject(value)) {
+      throw UnsupportedError('Invalid data type for boolean preparation', { type: entityType, attribute, value });
+    }
+    if (isEmptyField((value))) return '{}';
+    if (isObject(value)) {
+      return JSON.stringify(value);
+    }
+    // If string, check if json is valid
+    try {
+      JSON.parse(value);
+    } catch (e) {
+      throw UnsupportedError('Invalid data for json preparation', { type: typeof value, attribute: key, value });
+    }
+    return value.trim();
+  }
+  // Come from internal complex object only
+  if (attribute.type === 'object') { // For complex object, prepare inner elements
+    if (!isObject(value)) {
+      throw UnsupportedError('Invalid data type for object preparation', { type: entityType, attribute, value });
+    }
+    return value;
```

Comment:
Where do we prepare the inner elements of the object ?

Reply:
I remove that part for now has inner object elements are not defined by schema yet.